### ISSUE TITLE
Airflow 2.0.0 Azure AD Integration Not working

### ISSUE THREAD

**Issue by HariAA** [airflow-helm/charts#87]
Hi All

We are setting up airflow 2.0.0 on Azure Kubernetes Service and would like to configure azure ad integration for the webserver. 
I have done the changes to the webserver config like below. I am able to get the login option when navigating to webserver but i receive an invalid login error after signing in. Looking into the detailed logging from flask app builder, it is throwing an error saying "Error returning OAuth user info: 'upn'".
Has anyone faced this issue ?

    import os
    from airflow.configuration import conf
    from flask_appbuilder.security.manager import AUTH_OAUTH
    SQLALCHEMY_DATABASE_URI = conf.get("core", "SQL_ALCHEMY_CONN")
    basedir = os.path.abspath(os.path.dirname(__file__))
    CSRF_ENABLED = True
    AUTH_TYPE = AUTH_OAUTH
    AUTH_USER_REGISTRATION_ROLE = "Admin"
    OAUTH_PROVIDERS = [{
        'name':'azure',
        'token_key':'access_token',
        'icon':'fa-windows',
        'remote_app': {
            'base_url': 'https://login.microsoftonline.com/<TenantID>/oauth2',
            'request_token_params' :{'scope': 'User.read name preferred_username email profile upn'},
            'access_token_url':'https://login.microsoftonline.com/<TenantID>/oauth2/token',
            'authorize_url':'https://login.microsoftonline.com/<TenantID>/oauth2/authorize',
            'request_token_url': None,
            'client_id': '<Client_ID>',
            'client_secret': '<Client_Secret>'
        }
    }]

**Comment by kapopken** [comment]
@HariAA  try this
 
"remote_app": {
            "client_id": "<Client_ID>",
            "client_secret":  "<Client_Secret>",
            "api_base_url": "https://login.microsoftonline.com/<TenantID>/oauth2",
            "client_kwargs": {
                "scope": "User.read name preferred_username email profile upn",
                "resource": "<Client_ID>",
            },
            "request_token_url": None,
            "access_token_url": "https://login.microsoftonline.com/<TenantID>/oauth2/token",
            "authorize_url": "https://login.microsoftonline.com/<TenantID>/oauth2/authorize",
        },

**Comment by kapopken** [comment]
For everyone else looking to solve for this here are the steps.

1. Create an app registration in AzureAD (Dont forget to set the reply url to your hosted airflow)
2. Create a config map like this
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: my-webserver-config
  labels:
    app: airflow
data:
  webserver_config.py: |
    import os
    from airflow.configuration import conf
    from flask_appbuilder.security.manager import AUTH_OAUTH
    SQLALCHEMY_DATABASE_URI = conf.get("core", "SQL_ALCHEMY_CONN")
    basedir = os.path.abspath(os.path.dirname(__file__))
    AUTH_USER_REGISTRATION_ROLE = "Admin"
    AUTH_TYPE = AUTH_OAUTH
    AUTH_ROLES_SYNC_AT_LOGIN = True
    AUTH_USER_REGISTRATION = True
    AZURE_TENANT_ID = os.environ.get("AZURE_TENANT_ID")
    API_BASE = f"https://login.microsoftonline.com/{AZURE_TENANT_ID}/oauth2"
    ACCESS_TOKEN_URL = f"{API_BASE}/token"
    AUTHORIZE_URL = f"{API_BASE}/authorize"
    OAUTH_PROVIDERS = [
       {
        "name": "azure",
        "icon": "fa-windows",
        "token_key": "access_token",
        "remote_app": {
            "client_id": os.environ.get("AZURE_CLIENT_ID"),
            "client_secret":  os.environ.get("AZURE_CLIENT_SECRET"),
            "api_base_url": API_BASE,
            "client_kwargs": {
                "scope": "User.read name preferred_username email profile upn",
                "resource": os.environ.get("AZURE_CLIENT_ID"),
            },
            "request_token_url": None,
            "access_token_url": ACCESS_TOKEN_URL,
            "authorize_url": AUTHORIZE_URL,
        },
    }
    ]
```
3. Install the addition piplibs  
```
airflow.extraPipPackages:
  - "Flask-OpenID"
  - "authlib"
```
4. Mount the file 
 ```
airflow:
  extraConfigmapMounts:
    - name: my-webserver-config
      mountPath: /opt/airflow/webserver_config.py
      configMap: my-airflow-webserver-config
      readOnly: true
      subPath: webserver_config.py
```
5. Deploy

**Comment by HariAA** [comment]
@kapopken Thanks for sharing the steps. It didn't work with your config as well. BUt I figured it out. Turns out I had to enable the optional claim feature to add upn post which I was able to authenticate.

App Registration-->Token Registration: Add Optional Claim--> Select type as ID Token and field upn and set optional setting "externally authenticated" to Yes. that did the trick.

But thanks for your detailed steps. i would have saved 2 days had I received this earlier :)
